topics/basics/testing_plugins/integration_tests/integration_tests_ui.md

```diff
@@ -0,0 +1,198 @@
+<!-- Copyright 2000-2025 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+# Integration Tests: UI Testing
+
+<link-summary>Walkthrough how to interact with UI in integration tests.</link-summary>
+
+> This page is part of [](integration_tests.md) tutorial.
+
+For introduction and setting up dependencies please refer to [](integration_tests_intro.md).
+
+## UI Hierarchy
+
+IntelliJ-based IDEs primarily use Swing and AWT for their user interface, while [JCEF](embedded_browser_jcef.md) is used in specific cases like Markdown rendering.
+These UI frameworks organize elements in a parent-child hierarchy, similar to HTML's DOM structure:
+
+* Top-level containers (IDE frame and dialogs).
+* Nested containers.
+* Individual components (buttons, text fields, and lists).
+
+Every UI element (except top-level containers) must have a parent container, creating a clear hierarchical structure.
+
+The Driver framework provides a Kotlin DSL that mirrors this hierarchy.
+Here's an example:
+
+```kotlin
+ideFrame {
+  invokeAction("SearchEverywhere")
+  searchEverywherePopup {
+    actionButtonByXpath(xQuery { byAccessibleName("Preview")}).click()
+  }
+}
+```
+
+This code demonstrates hierarchical navigation:
+
+* Find the main IDE window (`ideFrame`).
+* Trigger the _Search Everywhere_ action (`invokeAction("SearchEverywhere")`).
+* Locate the <control>Search Everywhere</control> popup (`searchEverywherePopup`).
+* Find and click the <control>Preview</control> button within the popup (`actionButtonByXpath(xQuery { byAccessibleName("Preview")}).click()`).
+
+The code could be written more concise:
```

Comment:
```suggestion
The code could be more concise:
```

or:
`The code could be written more concisely:`
but I think the first one reads better.